### DISCUSSION TITLE
Bypass youtube region restriction

### DISCUSSION THREAD

**Discussion by KiyuM1n** [ReVanced/revanced-discussions#163] [Question]
some video and music have region restriction
maybe a mod for youtube and YTMusic to bypass that?

**Comment by oSumAtrIX** [comment]
Is that even possible?

**Comment by KiyuM1n** [comment]
youtube and YTMusic can use VPN to do it
but if you have youtube premium YTMusic is locked in the region where you purchase it

**Comment by oSumAtrIX** [comment]
In that case it might be impossible

**Comment by ArjixWasTaken** [comment]
> Is that even possible?

Yes, using a proxy that is, there is a userscript that does this for the browser, lemme find it real quick

**Comment by ArjixWasTaken** [comment]
Although this is for age restriction bypassing [which should also be implemented :)]

https://github.com/zerodytrash/Simple-YouTube-Age-Restriction-Bypass

It does have A LOT of code to read thru, I would suggest that we contact it's author, who knows, he might even help.

**Comment by oSumAtrIX** [comment]
In that case you should simply use the proxy in your phone

**Comment by ArjixWasTaken** [comment]
> In that case you should simply use the proxy in your phone

Nope, that's not how it works.
It is not simply the proxy that solves this issue. It is more like half of the solution.

I am not deeply informed on the subject, but the userscript I sent takes advantage of the YouTube embeds, YouTube embeds act a bit differently from the YouTube site itself. It is like some oversight by YouTube.

Anyways, for region bypassing yeah a proxy/VPN would work, but for age restriction bypassing it wouldn't.

Which, now that I think about it, might be what you meant, since age restriction bypassing was just a suggestion in this issue by me and not this issue's topic. If that is the case then please ignore me as I'm dumb.

**Comment by MulverineX** [comment]
Could use invidious mirrors to do this

**Comment by oSumAtrIX** [comment]
https://github.com/ReVanced/revanced-discussions/discussions/118
